dev/bots/test.dart

```diff
@@ -876,122 +705,294 @@ Future<void> _runFlutterTest(String workingDirectory, {
   }
 }
 
-// the optional `file` argument is an override for testing
-@visibleForTesting
-Future<bool> verifyVersion(String filename, [File file]) async {
-  final RegExp pattern = RegExp(r'^\d+\.\d+\.\d+(\+hotfix\.\d+)?(-pre\.\d+)?$');
-  file ??= File(filename);
-  final String version = await file.readAsString();
-  if (!file.existsSync()) {
-    print('$redLine');
-    print('The version logic failed to create the Flutter version file.');
-    print('$redLine');
-    return false;
-  }
-  if (version == '0.0.0-unknown') {
-    print('$redLine');
-    print('The version logic failed to determine the Flutter version.');
-    print('$redLine');
-    return false;
-  }
-  if (!version.contains(pattern)) {
-    print('$redLine');
-    print('The version logic generated an invalid version string: "$version".');
-    print('$redLine');
-    return false;
-  }
-  return true;
+Map<String, String> _initGradleEnvironment() {
+  final String androidSdkRoot = (Platform.environment['ANDROID_HOME']?.isEmpty ?? true)
+      ? Platform.environment['ANDROID_SDK_ROOT']
+      : Platform.environment['ANDROID_HOME'];
+  if (androidSdkRoot == null || androidSdkRoot.isEmpty) {
+    print('${red}Could not find Android SDK; set ANDROID_SDK_ROOT (or ANDROID_HOME).$reset');
+    exit(1);
+  }
+  return <String, String>{
+    'ANDROID_HOME': androidSdkRoot,
+    'ANDROID_SDK_ROOT': androidSdkRoot,
+  };
 }
 
-Future<void> _runIntegrationTests() async {
-  final String subShard = Platform.environment['SUBSHARD'];
-
-  switch (subShard) {
-    case 'gradle1':
-    case 'gradle2':
-      // This runs some gradle integration tests if the subshard is Android.
-      await _androidGradleTests(subShard);
-      break;
-    default:
-      await _runDevicelabTest('dartdocs');
-
-      if (Platform.isLinux) {
-        await _runDevicelabTest('flutter_create_offline_test_linux');
-      } else if (Platform.isWindows) {
-        await _runDevicelabTest('flutter_create_offline_test_windows');
-      } else if (Platform.isMacOS) {
-        await _runDevicelabTest('flutter_create_offline_test_mac');
-        await _runDevicelabTest('plugin_lint_mac');
-// TODO(jmagman): Re-enable once flakiness is resolved.
-//        await _runDevicelabTest('module_test_ios');
-      }
+final Map<String, String> gradleEnvironment = _initGradleEnvironment();
```

Comment:
cc @tvolkert @blasten this block is new, it's the result of merging with @blasten's recent changes

Basically I took the same approach as with the web tests (though sadly it's different enough that I couldn't reuse code).